libbeat/publisher/queue/memqueue/ringbuf.go

```diff
@@ -271,11 +204,16 @@ func (b *ringBuffer) Full() bool {
 	if b.regB.size > 0 {
 		avail = b.regA.index - b.regB.index - b.regB.size
 	} else {
-		avail = b.buf.Len() - b.regA.index - b.regA.size
+		// This doesn't seem right -- this checks how much space
+		// is available after region A in the internal array, but
+		// there might also be free space before region A. In that
+		// case new events must be inserted in region B, but the
+		// queue isn't at capacity.
+		avail = len(b.entries) - b.regA.index - b.regA.size
```

Comment:
What is the follow up from your comment about this possibly not being right? Is it too much work to fix, or not worth fixing?

Reply:
Hah -- this turned out to be the cause of the test failure 😅 This is the same computation as the old version, but before it was only made on a specific state transition, and now that its checked every loop iteration it ended up blocking the queue. Switching to the correct calculation here makes the tests pass locally, so fingers crossed on the CI now.